### DISCUSSION TITLE
Possible Accept Headers out of date for Several Engines (triage)

### DISCUSSION THREAD

**Discussion by glanham-jr** [searx/searx#3476] [Public Searx instances: report a bug]
This [issue](https://github.com/searx/searx/issues/3466) for Bing resulted in using an Accept Header that wasn't valid.

juusoohtonen mentioned that DDG is also not returning results. With this, I'll use this discussion for finding the impact of each engine that might be affected by this. Its a pretty simple fix, so shouldn't be too bad.

We should try to query each engine currently in Searx individually and see which ones are not returning results, and then see if a simple Accept Header update will fix it.

**Comment by juusoohtonen** [comment]
Can you easily check 1) which engines are broken at the moment and 2) which engines would get fixed with the simple Accept Header addition?
Do you have a script and/or test automation for validating different search engines?

**Comment by glanham-jr** [comment]
Sadly I don't have a script.

What I'm thinking of doing is adding a test(s), skipped by default (or only warn on failure and not fail the test suite), which one-by-one calls each engine with a query that should provide results to see if we can get at least one result. I would consider this an integration test, so I'll likely place it into a new folder named "integration" in the test path. This should greatly assist us now, and anyone in the future adding new engines.

**Comment by glanham-jr** [comment]
@juusoohtonen I have a [Draft PR](https://github.com/searx/searx/pull/3483) that includes integration tests based on set of engines. Below is an image of a sample output of the integration tests.

![image](https://user-images.githubusercontent.com/58887802/223906075-54362b90-db21-41a0-874d-58e606ad7c60.png)
